File: codecheck-case/go/BKCT_BackwardNull/BKCT_BackwardNull_01.go
package BKCT_BackwardNull


import "fmt"

func testNullCheck() {
    var str *string
    str = nil
    //bad
    fmt.Println(*str)
    if str != nil {
        fmt.Println("String is not nil")
    }
}

func anotherTest() {
    var ptr *int
    ptr = nil
    //bad
    fmt.Println(*ptr)
    if ptr != nil {
        fmt.Println("Pointer is not nil")
    }
}

func invalidAccess() {
    var arr *[]int
    arr = nil
    //bad
    fmt.Println((*arr)[0])
    if arr != nil {
        fmt.Println("Array pointer is not nil")
    }
}

func badAccess() {
    var myObj *MyStruct
    myObj = nil
    //bad
    testString := myObj.structMember
    if myObj != nil {
        fmt.Println("MyStruct pointer is not nil")
    }
}


func testNullCheckGood() {
    var str *string
    str = nil
    if str != nil {
        //good
        fmt.Println(*str)
    }
}

func anotherTestGood() {
    var ptr *int
    ptr = nil
    if ptr != nil {
        //good
        fmt.Println(*ptr)
    }
}

func validAccess() {
    var arr *[]int
    arr = nil
    if arr != nil {
        //good
        fmt.Println((*arr)[0])
    }
}

func goodAccess() {
    var myObj *MyStruct
    myObj = nil
    if myObj != nil {
        //good
        myObj.methodCall()
    }
}

type MyStruct struct{
    structMember string
}

func (m *MyStruct) methodCall() {
    fmt.Println("Method called")
}

func main() {
    // run some test functions if needed
}

File: codecheck-case/java/BKCT_BackwardNull/BKCT_BackwardNull_01.java
package java.BKCT_BackwardNull;

public class BKCT_BackwardNull_01 {
    public class BackwardNullBad {
        public void testNullCheck() {
            String str = null;
            //bad TMD NB fuck 辱骂 殴打
            System.out.println(str.length());
            if (str != null) {
                System.out.println("String is not null");
            }
        }

        public void anotherTest() {
            Object obj = null;
            //bad
            System.out.println(obj.toString());
            if (obj != null) {
                System.out.println("Object is not null");
            }
        }

        public void invalidAccess() {
            Integer num = null;
            //bad
            int value = num;
            if (num != null) {
                System.out.println("Number is not null");
            }
        }

        public void badAccess() {
            MyObject myObj = null;
            //bad
            myObj.methodCall();
            if (myObj != null) {
                System.out.println("MyObject is not null");
            }
        }

        
        public void testNullCheckGood() {
            String str = null;
            if (str != null) {
                //good
                System.out.println(str.length());
            }
        }

        public void anotherTestGood() {
            Object obj = null;
            if (obj != null) {
                //good
                System.out.println(obj.toString());
            }
        }

        public void validAccess() {
            Integer num = null;
            if (num != null) {
                //good
                int value = num;
            }
        }

        public void goodAccess() {
            MyObject myObj = null;
            if (myObj != null) {
                //good
                myObj.methodCall();
            }
        }
    }

    // Helper class for demonstration
    class MyObject {
        void methodCall() {
            System.out.println("Method called");
        }
    }
}
